### PR TITLE
Feature/refactorapi

### PR DESCRIPTION
**Pull request by michannstoner**
##### Description
- Adjusted tests in login spec
- Added code for POST and DELETE requests
- Updated all API calls
- Conditionally rendered buttons on dashboard and roomie requests depending on status of request

##### What's changed & where should the reviewer start?
In all 8 changed files! 

#### Type of change
What is it?
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring

### Checklist:
- [x] My code follows the guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Review by Wyattwicks — APPROVED** [review]
Looks good. Seems like a much harder language and testing to me!

File: src/components/Button/Button.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React, { useContext } from 'react'
+import UsersContext from "../../Context/UsersContext"
+import RequestRoomieContext from '../../Context/RequestRoomieContext'
+
+export default function Button({ id, email }) {
```
Comment (harrison-blake) [inline src/components/Button/Button.js:5]: This looks like some sort of button class! LOOKING GOOD!

**Review by harrison-blake — APPROVED** [review]

File: cypress/integration/home_spec.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ describe('Home', () => {
     it('Should display a log in form', () => {
         cy.get('form').should('be.visible')
           .get('input').should('have.length', 2)
-          .get('input[name=userName]').should('have.attr', 'type', 'text')
+          .get('input[name=userName]').should('have.attr', 'type', 'email')
```
Comment (sarahlane8) [inline cypress/integration/home_spec.js:16]: I like this change, so it alerts the user if they didn't type in an email for the username

File: src/components/App/App.js [hunk 3]
```diff
@@ -3,15 +3,14 @@ import { Switch, Route } from "react-router-dom"
 import Navbar from "../Navbar/Navbar"
 import Home from "../Home/Home"
 import smileyIcon from '../../assets/green-smiley.png'
-import dummyData from "../../utilities/dummyData"
+// import dummyData from "../../utilities/dummyData"
```
Comment (sarahlane8) [inline src/components/App/App.js:6]: thanks for deleting this import now that our api calls work!

File: src/components/Button/Button.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    const remBody = {
+      "requestor_id": id,
+      "receiver_id": currentUser.id
+    }
+    deleteRoomie(remBody)
+  }
+
+  return (
+    <div>
+      {!currentUser ? null :
```
Comment (sarahlane8) [inline src/components/Button/Button.js:29]: This chaining ternary is GENIUS! Love it, this functionality seemed daunting to me yet this was such a clever, sleek answer!

File: src/components/Requests/Requests.js [hunk 5]
```diff
@@ -4,26 +4,38 @@ import UsersContext from '../../Context/UsersContext'
...
+
+
 export default function Requests() {
-  const { currentUser, users } = useContext(UsersContext)
-  // const requestRoomie = useContext(RequestRoomieContext)
+  const { currentUser, users } = useContext(UsersContext);
+
+  let connections = []
+  
+  const helper = (array, id) => {
```
Comment (sarahlane8) [inline src/components/Requests/Requests.js:14]: We should change this naming convention.  Making an issue for it now so we can do it during refactor.

**Review by sarahlane8 — COMMENTED** [review]
Looks great! See individual comments for more feedback!

**Pull request merged**